torchtune/datasets/_instruct.py

```diff
@@ -124,73 +134,120 @@ def instruct_dataset(
+            and "output" column names to the actual column names in the dataset. Keys should be "input" and
+            "output" and values should be the new column names. Default is None, keeping the default "input"
+            and "output" column names.
+        train_on_input (bool): Whether the model is trained on the user prompt or not.
+            Default is False.
+        packed (bool): Whether or not to pack the dataset to tokenizer's ``max_seq_len`` prior to training. Default is False.
         **load_dataset_kwargs (Dict[str, Any]): additional keyword arguments to pass to ``load_dataset``,
             such as ``data_files`` or ``split``.
 
     Examples:
+
+        my_dataset.json
+        [
+            {
+                "question": "What time is it in London?",
+                "answer": "It is 10:00 AM in London.",
+            },
+            {
+                ...
+            },
+            ...,
+        ]
+
+    ::
+
         >>> from torchtune.datasets import instruct_dataset
         >>> dataset = instruct_dataset(
-        ...   tokenizer=tokenizer,
-        ...   source="yahma/alpaca_cleaned",
-        ...   template="torchtune.data.AlpacaInstructTemplate",
-        ...   max_seq_len=2096,
-        ...   train_on_input=True,
-        ...   packed=True,
+        ...     tokenizer=tokenizer,
+        ...     source="json",
+        ...     data_files="my_dataset.json",
+        ...     column_map={
+        ...         "input": "question",
+        ...         "output": "answer",
+        ...     },
+        ...     train_on_input=False,
+        ...     packed=False,
+        ...     split="train",
         ... )
+        >>> tokens = dataset[0]["tokens"]
+        >>> tokenizer.decode(tokens)
+        "What time is it in London?It is 10:00 AM in London."
 
     This can also be accomplished via the yaml config::
 
         dataset:
             _component_: torchtune.datasets.instruct_dataset
-            source: yahma/alpaca_cleaned
-            template: torchtune.data.AlpacaInstructTemplate
-            max_seq_len: 2096
-            train_on_input: True
-            packed: True
+            source: json
+            data_files: my_dataset.json
+            column_map:
+              input: question
+              output: answer
+            train_on_input: False
+            packed: False
+            split: train
 
     Returns:
-        InstructDataset or PackedDataset: the configured :class:`~torchtune.datasets.InstructDataset`
+        SFTDataset or PackedDataset: the configured :class:`~torchtune.datasets.SFTDataset`
             or :class:`~torchtune.datasets.PackedDataset` if ``packed=True``
+
+    Raises:
+        ValueError: If ``packed=True`` and ``tokenizer.max_seq_len`` is not set.
     """
-    ds = InstructDataset(
-        tokenizer=tokenizer,
+
+    message_transform = InputOutputToMessages(
+        train_on_input=train_on_input, column_map=column_map
+    )
```

Comment:
Did we discuss renaming this or did I imagine that? Also I'm a bit out of the loop on #1366 but why do we use a single static system prompt in `InputOutputToMessages` now? Is that sufficient? (Sorry if you guys already hashed this all out)

Reply:
Don't recall that discussion but @joecummings was interested in potentially renaming this.

As for the system prompt, yes a single prompt is sufficient to add a system message for every sample conversation.